### PR TITLE
kitsas: 4.0.5 -> 5.7

### LINKED ISSUE #346448: Update request: kitsas 4.0.5 → 5.7
- Package name: Kitsas
- Latest released version: 5.7

- Current version on the unstable channel: 4.0.5
- Current version on the stable/release channel: 4.0.5

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**
@gspia 

The current version in repo is not usable as many details have changed and rendered otherwise working application pretty much unusable. I've managed to do some research to make v5.7  ready for PR, but I can't make it build. qtwebengine must be added to arguments and nativeBuildInputs to clear some dependencies.

-----

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by itepastra**
Updated kitsas to version 5.7, 
also updated it to use qt6 instead of qt5 due to a function needed from qt6's qtpdf module
removed a replace since it didn't exist in the source anymore

fixes: #346448

I tested running the build result and it started correctly.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

I've opened kitsas from the command line. 
It looked similar to what it looked like before the changes. 
I don't understand Swedish or Finnish, so I cannot navigate deeply into the program.

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by gspia** [comment]
Thank you very much @itepastra : I missed that little detail on how to change it to use qt6. 
While trying this out, I made a flake for the "kitupiikki" or "kitsas" (this might be usable for those who are interested more deeply in this or sometimes for some reason need a version that is not available on nixpkgs).

```
# Get the sources from https://github.com/artoh/kitupiikki
# Modify paketoi-Linux-qt6.sh file in the following way:
# - comment QMAKE-line
# - comment last lines from wget-things
# Then, with this flake:
#  nix develop -c zsh
#  ./paketoi-Linux-qt6.sh
# The compilation should succeeds and after that the sw should work
#  ./build-linux-qt6/kitsas
{
  inputs.nixpkgs.url = "github:nixos/nixpkgs";

  outputs = { self, nixpkgs }:
  let
    pkgs = nixpkgs.legacyPackages.x86_64-linux;
  in {
    devShells.x86_64-linux.default = pkgs.mkShell {
      buildInputs = with pkgs; [
        # cmake
        # gdb
        qt6.qtbase
        qt6.full
        # qtcreator

        # this is for the shellhook portion
        qt6.wrapQtAppsHook
        # makeWrapper
        bashInteractive # without this, the shell feels odd..

        # 
        # qtwayland
        kdePackages.qtwayland
        # qt5compat
        kdePackages.qt5compat
        # qtsvg
        kdePackages.qtsvg
        libzip
        poppler
      ];
      # set the environment variables that Qt apps expect
      shellHook = ''
        export QT_QPA_PLATFORM=wayland
        bashdir=$(mktemp -d)
        makeWrapper "$(type -p bash)" "$bashdir/bash" "''${qtWrapperArgs[@]}"
        exec "$bashdir/bash"
      '';
    };
  };
}
```

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -30492,7 +30492,7 @@ with pkgs;
 
   kile = callPackage ../applications/editors/kile { };
 
-  kitsas = libsForQt5.callPackage ../applications/office/kitsas { };
+  kitsas = qt6Packages.callPackage ../applications/office/kitsas { };
```
Comment (Atemu) [inline pkgs/top-level/all-packages.nix:30495]: I'm not knowledgeable here but I think the preferred pattern these days is to use regular `callPackage` and just depend on `qt6`. You then take the relevant deps out of that package set.

That'd also allow moving this package to by-name.

**Review by Atemu — APPROVED** [review]
Diff generally LGTM but I don't have much experience with QT.

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (itepastra) [reply pkgs/top-level/all-packages.nix:30495]: I've made it depend on qt6 instead, moved it to by-name and I've optionally enabled qtwayland for linux.

**Pull request merged**